### ISSUE TITLE
Windows Compatible ??

### ISSUE THREAD

**Issue by lostSail0r** [Y2Z/monolith#54]
Forgive me if I'm being silly, but can this work for Windows CMD ?

**Comment by vflyson** [comment]
It should work, we have it successfully building on windows in AppVeyor.

**Comment by KurtDeGreeff** [comment]
Would you please be so kind to publish the windows binaries?

**Comment by snshn** [comment]
I'll get a VM and build it for Windows after that, will notify in this thread once it's done.

**Comment by fredimachado** [comment]
I've tested it using Windows Subsystem for Linux (WSL). You just need to install WSL, open it and execute the following commands:
```
$ sudo apt-get update
$ sudo apt-get upgrade
$ sudo apt-get install gcc pkg-config libssl-dev -y
```
The commands above will take a long time to run, I'll suggest you to grab some coffee :D

After it's done you can cd to your folder (I've executed `$ cd ~` to use my home folder), you can follow the monolith instructions. And generate your HTML.

If you want to open the current folder in Windows explorer run this:

`$ explorer.exe .`

Now you can open the HTML file in your browser. :D

Thanks a lot @snshn! This tool is amazing!

**Comment by sogaiu** [comment]
FWIW, building and running worked here without WSL.

**Comment by Tragen** [comment]
Can you share the binaries?

**Comment by sogaiu** [comment]
I don't know if it needs any extra dlls, but you can find a copy of at least the .exe I got at:

https://github.com/sogaiu/monolith/raw/master/monolith.exe

The sha256 checksum I got for it is:

6eb844d63ae787086964c17390917e5e0640b6c1ada7e04e578c1dec0a3ea1c5

**Comment by Tragen** [comment]
It works. Thank you so much.
How did you build it on Windows? Do you have a batch to compile it? What did you do?

**Comment by sogaiu** [comment]
Glad it worked :)

The command I used to compile was `cargo build --release` via `cmd.exe`.

It's possible that the toolchain used may be relevant.  At some point it had the setting `stable-x86_64-pc-windows-gnu`, and unfortunately, this didn't work in another project.  So I set it to: `stable-x86_64-pc-windows-msvc` using the `rustup toolchain` subcommand [1].  That seemed to help so I left it at that value.

(The machine I did the build on had other compiler-related bits on it already for other non-rust projects, I'm not sure if those portions got used in the comilation of monolith.)

[1] https://github.com/rust-lang/rustup#toolchain-specification
